.github/workflows/tests.yml

```diff
@@ -0,0 +1,90 @@
+name: Tests
+
+on:
+    push:
+    pull_request:
+
+jobs:
+
+    integration:
+        name: Integration
+        runs-on: ubuntu-latest
+
+        strategy:
+            matrix:
+                php: ['7.1', '7.4']
+
+        services:
+            redis:
+                image: redis:6.0.0
+                ports:
+                    - 6379:6379
+            redis-cluster:
+                image: grokzen/redis-cluster:5.0.4
+                ports:
+                    - 7000:7000
+                    - 7001:7001
+                    - 7002:7002
+                    - 7003:7003
+                    - 7004:7004
+                    - 7005:7005
+                    - 7006:7006
+                    - 7007:7007
+                env:
+                    STANDALONE: true
+            memcached:
+                image: memcached:1.6.5
+                ports:
+                    - 11211:11211
+            rabbitmq:
+                image: rabbitmq:3.8.3
+                ports:
+                    - 5672:5672
+
+        steps:
+            -   name: Checkout
+                uses: actions/checkout@v2
+
+            -   name: Setup PHP
+                uses: shivammathur/setup-php@v2
+                with:
+                    coverage: "none"
+                    extensions: "memcached,redis,xsl"
+                    ini-values: "memory_limit=-1"
+                    php-version: "${{ matrix.php }}"
+
+            -   name: Configure composer
+                run: |
+                    ([ -d ~/.composer ] || mkdir ~/.composer) && cp .github/composer-config.json ~/.composer/config.json
+                    SYMFONY_VERSION=$(cat composer.json | grep '^ *\"dev-master\". *\"[1-9]' | grep -o '[0-9.]*')
+                    echo "::set-env name=SYMFONY_VERSION::$SYMFONY_VERSION"
+                    echo "::set-env name=COMPOSER_ROOT_VERSION::$SYMFONY_VERSION.x-dev"
+
+            -   name: Determine composer cache directory
+                id: composer-cache
+                run: echo "::set-output name=directory::$(composer config cache-dir)"
+
+            -   name: Cache composer dependencies
+                uses: actions/cache@v1
+                with:
+                    path: ${{ steps.composer-cache.outputs.directory }}
+                    key: ${{ matrix.php }}-composer-${{ hashFiles('**/composer.lock') }}
+                    restore-keys: ${{ matrix.php }}-composer-
+
+            -   name: Install Symfony Flex
+                run: composer global require --no-progress --no-scripts --no-plugins symfony/flex dev-master
```

Comment:
Flex can be included as a "tool" in shivammathur/setup-php@v2

https://github.com/shivammathur/setup-php#wrench-tools-support